### PR TITLE
Add utility to retrieve user-specific configuration file directory

### LINKED ISSUE #61: Implement `configdir`
See [utils-configdir](https://github.com/kgryte/utils-configdir). `@stdlib/utils/configdir`.

### PR DESCRIPTION
**Pull request by Planeshifter**
Resolves #61.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

* [x] Read, understood, and followed the [contributing guidelines][contributing], including the relevant style guides.
* [x] Read and understand the [Code of Conduct][code-of-conduct].
* [x] Read and understood the [licensing terms][license].
* [x] Searched for existing issues and pull requests __before__ submitting this pull request.
* [x] Filed an issue (or an issue already existed) __prior to__ submitting this pull request.
* [x] Rebased onto latest `develop`.
* [x] Submitted against `develop` branch.

## Description

> What is the purpose of this pull request?

This pull request:

* contains a straight-forward port of [utils-configdir](https://github.com/kgryte/utils-configdir)

## Related Issues

> Does this pull request have any related issues?

This pull request:

* resolves #
* fixes #

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

---

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/master/CONTRIBUTING.md
[code-of-conduct]: https://github.com/stdlib-js/stdlib/blob/master/CODE_OF_CONDUCT.md
[license]: https://github.com/stdlib-js/stdlib/blob/master/LICENSE

### CODE REVIEW

File: lib/node_modules/@stdlib/utils/configdir/lib/configdir.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+var homedir = require( '@stdlib/utils/homedir' );
+
+
+// MAIN //
+
+/**
+* Returns a directory for user-specific configuration files.
+*
+* @param {string} [p] - path to append to a base directory
+* @returns {(string|null)} directory
```
Comment (kgryte) [inline lib/node_modules/@stdlib/utils/configdir/lib/configdir.js:19]: Add `throws` annotation.

File: lib/node_modules/@stdlib/utils/configdir/lib/configdir.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
+'use strict';
+
+// MODULES //
+
+var path = require( 'path' );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/utils/configdir/lib/configdir.js]: `var join = require( 'path' ).join;`

This change will affect subsequent usage below.

File: lib/node_modules/@stdlib/utils/configdir/README.md [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+```
+
+To append a `path` to the base directory, provide a `path` argument.
+
+``` javascript
+var dir = configdir( 'appname/config' );
+// e.g., returns '/Users/<username>/Library/Preferences/appname/config'
+```
+
+On non-windows platforms, if the function is unable to locate the current user's [`home`][@stdlib/utils/homedir] directory, the function returns `null`.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/utils/configdir/README.md]: We should probably capitalize Windows here => `non-Windows`. Same goes for similar usage in `./test/test.js`.

Comment (kgryte) [reply lib/node_modules/@stdlib/utils/configdir/README.md]: Should also add

> Similarly, on Windows platforms, if the function is unable to locate an application data directory, the function also returns `null`.

**Review by kgryte — CHANGES_REQUESTED** [review]

**Comment by Planeshifter** [comment]
Thanks! Made the requested changes.

**Review by kgryte — APPROVED** [review]

**Pull request merged**